templates/eap-cd-starter-s2i.json

```diff
@@ -0,0 +1,342 @@
+            "description": "List of directories from which archives will be copied into the deployment folder. If unspecified, all archives in /target will be copied.",
+            "name": "ARTIFACT_DIR",
+            "value": "",
+            "required": false
+        },
+        {
+            "description": "Container memory limit",
+            "name": "MEMORY_LIMIT",
+            "value": "1Gi",
+            "required": false
+        }
+    ],
+    "objects": [
+        {
+            "kind": "Service",
+            "apiVersion": "v1",
+            "spec": {
+                "ports": [
+                    {
+                        "port": 8080,
+                        "targetPort": 8080
+                    }
+                ],
+                "selector": {
+                    "deploymentConfig": "${APPLICATION_NAME}"
+                }
+            },
+            "metadata": {
+                "name": "${APPLICATION_NAME}",
+                "labels": {
+                    "application": "${APPLICATION_NAME}"
+                },
+                "annotations": {
+                    "description": "The web server's http port."
+                }
+            }
+        },
+        {
+            "kind": "Route",
+            "apiVersion": "v1",
+            "id": "${APPLICATION_NAME}-https",
+            "metadata": {
+                "name": "${APPLICATION_NAME}",
+                "labels": {
+                    "application": "${APPLICATION_NAME}"
+                },
+                "annotations": {
+                    "description": "Route for application's https service."
+                }
+            },
+            "spec": {
+                "to": {
+                    "name": "${APPLICATION_NAME}"
+                },
+                "tls": {
+                    "insecureEdgeTerminationPolicy": "Redirect",
+                    "termination": "edge"
+                }
+            }
+        },
+        {
+            "kind": "ImageStream",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "${APPLICATION_NAME}",
+                "labels": {
+                    "application": "${APPLICATION_NAME}"
+                }
+            }
+        },
+        {
+            "kind": "BuildConfig",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "${APPLICATION_NAME}",
+                "labels": {
+                    "application": "${APPLICATION_NAME}"
+                }
+            },
+            "spec": {
+                "source": {
+                    "type": "Git",
+                    "git": {
+                        "uri": "${SOURCE_REPOSITORY_URL}",
+                        "ref": "${SOURCE_REPOSITORY_REF}"
+                    },
+                    "contextDir": "${CONTEXT_DIR}"
+                },
+                "strategy": {
+                    "type": "Source",
+                    "sourceStrategy": {
+                        "env": [
```

Comment:
Thats not a bad idea, definitely a good idea for the pro versions. 

@bstansberry WDYT about SSO on starter?

Reply:
@luck3y Let's chat about it in the morning.

Reply:
@luck3y @rcernich Following our irc chat this morning I think this is out of scope for this PR.

If we say the User Story around this piece is "I want to spin up an EE app on OSO Starter that integrates with an RH-SSO server" then we'd need:

1) The RH-SSO server. That's not part of this template (isn't part of eap-xx-sso-s2i templates either; seems to be something assumed to be set up separately).  So user would have to set that up themselves, and if trying to do it on the same OSOS account as the EAP server they will likely hit memory limits etc that they'd have to work around.

2) Various params in this template matching those in eap-cd-sso-s2i that drive the integration with SSO.  But without the SSO server those are distractions to the user. We're eliminating distracting params in this template which is a main reason it exists instead of using eap-cd-basic-s2i.  I think the JGroups-related params we've eliminated are as likely to be useful on OSOS as SSO ones, probably more, as you could spin up a 2 node EAP cluster on OSOS easily enough by setting MEMORY_LIMIT to 500MB.

So, I don't see a User Story we achieve by doing this.

Now, all that said I'm fine with making a change like https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/97 for this template too if we decide that's a standard way we want our stuff to work. Then it's somewhat an internal detail of how we set up our containers, where we want to do it a standard way.  But that seems out of scope for this PR and something to do separately. At least after #97 is merged.